app/Http/Controllers/API/TorrentController.php

```diff
@@ -79,15 +81,21 @@ public function store(Request $request, Torrent $torrent)
         // Deplace and decode the torrent temporarily
         $decodedTorrent = TorrentTools::normalizeTorrent($requestFile);
         $infohash = Bencode::get_infohash($decodedTorrent);
-        $meta = Bencode::get_meta($decodedTorrent);
-        $fileName = uniqid().'.torrent'; // Generate a unique name
-        file_put_contents(getcwd().'/files/torrents/'.$fileName, Bencode::bencode($decodedTorrent));
+
+        try {
+            $meta = Bencode::get_meta($decodedTorrent);
+        } catch (\Exception $e) {
+            return $this->sendError('Validation Error.', 'You Must Provide A Valid Torrent File For Upload!');
+        }
+
+        $fileName = sprintf('%s.torrent', uniqid()); // Generate a unique name
+        Storage::disk('torrents')->put($fileName, Bencode::bencode($decodedTorrent));
 
         // Find the right category
         $category = Category::withCount('torrents')->findOrFail($request->input('category_id'));
 
         // Create the torrent (DB)
-        $torrent = new Torrent();
+        $torrent = app()->make(Torrent::class);
```

Comment:
Why the change here?

Reply:
It was a drive-by refactor. Generally speaking, creating a new instance with the `new` keyword should be avoided wherever possible in favor of resolving the class from the service container. And in fact, this should actually be `resolve(Torrent::class)`, which is short-hand for what's there now.

In this specific instance, the change does not affect functionality, but it makes the code more robust. Even though `Torrent` is "just a simple Eloquent model", and is unlikely to change or ever require a constructor argument, using the container like this adds the ability to hook into the service lifecycle and, for example, interact with a `Torrent` instance that is resolved from the container before it's given to its consumer.

In short, `resolve()` should always be favored over a plain-old `new ...`.

Reply:
Got it, thanks for explaining. Sounds like a global refactor for the future.